### PR TITLE
Carson/dfsan mods

### PR DESCRIPTION
**Pull request by carsonharmon**
Mods to DFSan for polytracker integration. 

Forking LLVM lets us pull upstream pretty easily, and debloats are codebase. Makes everything much easier to maintain.

### CODE REVIEW

File: compiler-rt/lib/dfsan/dfsan.cpp [hunk 1]
```diff
@@ -519,7 +606,10 @@ static void dfsan_init(int argc, char **argv, char **envp) {
   Atexit(dfsan_fini);
   AddDieCallback(dfsan_fini);
 
-  __dfsan_label_info[kInitializingLabel].desc = "<init label>";
+  //label_map_t::Handle new_label(&__dfsan_label_info, kInitializingLabel);
```
Comment (ESultanik) [inline compiler-rt/lib/dfsan/dfsan.cpp:609]: Why is this commented out?

File: compiler-rt/lib/dfsan/dfsan.cpp [hunk 2]
```diff
@@ -143,21 +163,39 @@ SANITIZER_INTERFACE_ATTRIBUTE uptr __dfsan_shadow_ptr_mask;
 // | reserved by kernel |
 // +--------------------+ 0x0000000000
 
-typedef atomic_dfsan_label dfsan_union_table_t[kNumLabels][kNumLabels];
-
+//typedef atomic_dfsan_label dfsan_union_table_t[kNumLabels][kNumLabels];
+typedef atomic_dfsan_label old_union_table_t[oldNumLabels][oldNumLabels];
```
Comment (ESultanik) [inline compiler-rt/lib/dfsan/dfsan.cpp:167]: This is going to get removed once we switch to the hash map, right?

**Review by ESultanik — APPROVED** [review]

File: compiler-rt/lib/dfsan/dfsan.cpp (hunk 1 above)
Comment (carsonharmon) [reply compiler-rt/lib/dfsan/dfsan.cpp:609]: This was my use of the hashtable I modified from the sanitizer lib, and was using it earlier. Instead, we are pushing this functionality into polytracker. So we no longer need it, its basically just dead code, we can remove it

File: compiler-rt/lib/dfsan/dfsan.cpp (hunk 2 above)
Comment (carsonharmon) [reply compiler-rt/lib/dfsan/dfsan.cpp:167]: this is actually not used except during address calculations for the mmaped region. We won't be using it at all, and instead using a hashtable within polytracker (just unordered map, or we can just use the taint forest)

File: compiler-rt/lib/dfsan/dfsan.cpp [hunk 3]
```diff
@@ -143,21 +163,39 @@ SANITIZER_INTERFACE_ATTRIBUTE uptr __dfsan_shadow_ptr_mask;
...
+//typedef atomic_dfsan_label dfsan_row[kNumLabels];
 #ifdef DFSAN_RUNTIME_VMA
 // Runtime detected VMA size.
 int __dfsan::vmaSize;
 #endif
 
 static uptr UnusedAddr() {
-  return UnionTableAddr() + sizeof(dfsan_union_table_t);
+  return UnionTableAddr() + sizeof(old_union_table_t);
```
Comment (carsonharmon) [inline compiler-rt/lib/dfsan/dfsan.cpp:175]: @ESultanik You can see here where the union table size is used to partition a region of the mmaped space. We technically don't need this space anymore, but it should not really affect us. If you look in dfsan_platform.h, I already bumped up the address space to support all the shadow memory we want.

**Pull request merged**